### PR TITLE
BREAKING CHANGE: replace `etl::smart_ptr` with `std::shared_ptr`

### PR DESCRIPTION
**Pull request by rodolforg**
- Remove `etl::smart_ptr`
- `synfig::SmartFILE` is now based on `std::shared_ptr`

### CODE REVIEW

**Comment by rodolforg** [comment]
Should this be labeled as `BREAKING CHANGE:` instead of `refactor?`

**Comment by ice0** [comment]
> Should this PR be labeled as `BREAKING CHANGE:` instead of `refactor`?

I think yes.

File: synfig-studio/src/gui/duckmatic.cpp [hunk 1]
```diff
@@ -2514,14 +2514,14 @@ Duckmatic::add_to_ducks(const synfigapp::ValueDesc& value_desc,etl::handle<Canva
...
-						etl::smart_ptr<synfig::Point> point(new Point(tangent1_duck->get_point()));
+						std::shared_ptr<synfig::Point> point(new Point(tangent1_duck->get_point()));
 						tangent1_duck->set_shared_point(point);
 						tangent2_duck->set_shared_point(point);
 					}
 					else
 					if (!bline_point.get_split_tangent_angle())
 					{
-						etl::smart_ptr<synfig::Angle> angle(new Angle(
+						std::shared_ptr<synfig::Angle> angle(new Angle(
```
Comment (ice0) [inline synfig-studio/src/gui/duckmatic.cpp:2524]: I suggest to replace `new` with `std::make_shared`

File: synfig-studio/src/gui/states/state_stroke.h [hunk 2]
```diff
@@ -57,13 +58,13 @@ extern StateStroke state_stroke;
...
-	etl::smart_ptr<std::list<synfig::Real> > width_data;
+	std::shared_ptr<std::list<synfig::Point>> stroke_data;
+	std::shared_ptr<std::list<synfig::Real>> width_data;
 	Gdk::ModifierType modifier;
 
-	EventStroke(etl::smart_ptr<std::list<synfig::Point> > stroke_data,
-			etl::smart_ptr<std::list<synfig::Real> > width_data,
-			Gdk::ModifierType modifier=Gdk::ModifierType(0)
+	EventStroke(std::shared_ptr<std::list<synfig::Point>> stroke_data,
+	        std::shared_ptr<std::list<synfig::Real>> width_data,
```
Comment (ice0) [inline synfig-studio/src/gui/states/state_stroke.h:66]: Spaces

File: synfig-studio/src/gui/duckmatic.cpp (hunk 1 above)
Comment (rodolforg) [reply synfig-studio/src/gui/duckmatic.cpp]: Done

File: synfig-studio/src/gui/states/state_stroke.h (hunk 2 above)
Comment (rodolforg) [reply synfig-studio/src/gui/states/state_stroke.h]: I think I changed how it was now

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**